storage-dynamodb/src/test/java/io/delta/storage/FailingDynamoDBLogStore.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.storage;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import java.io.IOException;
+import java.util.stream.Stream;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class FailingDynamoDBLogStore extends DynamoDBLogStore {
```

Comment:
where is this used?

Reply:
We can use this logstore implementation in our integration test to inject random errors in few places and test error handling / log fixing.